app/jobs/map_github_repositories_to_owners.py

```diff
@@ -1,5 +1,286 @@
+from app.main.services.database_service import DatabaseService
+import logging
+from app.main.config.app_config import app_config
+from app.main.config.logging_config import configure_logging
+from app.main.services.github_service import GithubService
+
+import json
+
+logger = logging.getLogger(__name__)
+
+
+def convert_to_percentage(partial: int, total: int) -> str:
+    return f"{round((partial / total) * 100)}%"
+
+
+def contains_one_or_more(values: list[str], list_to_check: list[str]) -> bool:
+    found = False
+
+    for value in values:
+        if value in list_to_check:
+            found = True
+
+    return found
+
+
 def main():
-    print("Running...")
+    configure_logging(app_config.logging_level)
+    logger.info("Running...")
+    database_service = DatabaseService()
+    github_service = GithubService(app_config.github.token)
+    repositories = github_service.get_all_repositories()
+
+    hmpps_owner_id, _ = database_service.find_owner_by_name("HMPPS")[0]
+    laa_owner_id, _ = database_service.find_owner_by_name("LAA")[0]
+    opg_owner_id, _ = database_service.find_owner_by_name("OPG")[0]
+    cica_owner_id, _ = database_service.find_owner_by_name("CICA")[0]
+    central_digital_owner_id, _ = database_service.find_owner_by_name(
+        "Central Digital"
+    )[0]
+    platforms_and_architecture_owner_id, _ = database_service.find_owner_by_name(
+        "Platforms and Architecture"
+    )[0]
+    tech_services_owner_id, _ = database_service.find_owner_by_name("Tech Services")[0]
+
+    strict_search = False
+    direct_access_level = (
+        "github_teams_with_admin_access"
+        if strict_search
+        else "github_teams_with_any_access"
+    )
+    parent_access_level = (
+        "github_teams_with_admin_access_parents"
+        if strict_search
+        else "github_teams_with_any_access_parents"
+    )
+
+    unownedRepos = []
+    reposWithMultipleOwners = []
+    hmppsRepos = []
+    laaRepos = []
+    opgRepos = []
+    cicaRepos = []
+    centralDigitalRepos = []
+    platformsAndArchitectureRepos = []
+    techServicesRepos = []
+
+    for repository in repositories:
+        ownersFound = 0
+        if (
+            # HMPPS Digital
+            "HMPPS Developers" in repository[direct_access_level]
+            or "HMPPS Developers" in repository[parent_access_level]
+            # Fuzzy Matches 👇
+            or repository["name"].startswith("hmpps-")
+        ):
+            database_service.add_relationship_between_asset_and_owner(
+                repository["name"], hmpps_owner_id
+            )
+            hmppsRepos.append(repository)
+            ownersFound += 1
+
+        if (
+            # LAA Digital
+            "LAA Technical Architects" in repository[direct_access_level]
```

Comment:
Could perhaps use `any`? Though might not make it any clearer. 🐜 🦇 🐱 🐶 


substring_list = ["ant", "bat", "cat", "dog"]
string = "the cat sat on the mat, an ant crawled by"
res = any(substring in string for substring in substring_list)
print(res) # True


substring_list = ["bat", "dog"]
string = "the cat sat on the mat, an ant crawled by"
res = any(substring in string for substring in substring_list)
print(res) # False